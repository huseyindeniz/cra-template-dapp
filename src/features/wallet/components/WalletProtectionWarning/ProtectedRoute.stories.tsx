// ConnectButton.stories.ts|tsx
import { ComponentStory, ComponentMeta } from "@storybook/react";

import { WalletProtectionWarning } from ".";

export default {
  title: "Components/WalletProtectionWarning",
  component: WalletProtectionWarning,
} as ComponentMeta<typeof WalletProtectionWarning>;

export const Default: ComponentStory<typeof WalletProtectionWarning> = (
  args
) => <WalletProtectionWarning />;
