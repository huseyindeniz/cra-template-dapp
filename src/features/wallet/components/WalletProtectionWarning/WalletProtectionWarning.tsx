import {
  Box,
  Center,
  Container,
  Stack,
  Heading,
  Divider,
} from '@chakra-ui/react';
import { useTranslation } from 'react-i18next';

import { AlertMessage } from '../../../ui/components/AlertMessage/AlertMessage';

export const WalletProtectionWarning: React.FC = () => {
  const { t } = useTranslation('FeatureWallet');
  return (
    <Box>
      <Container maxW="7xl" py={2} as={Stack} spacing={12}>
        <Stack spacing={0} align="center">
          <Heading mb={2}>{t('401 Unauthorized')}</Heading>
          <Divider />
        </Stack>
        <Center>
          <AlertMessage status="error" title="Access Denied">
            {t('You need to connect your wallet to see this page.')}
          </AlertMessage>
        </Center>
      </Container>
    </Box>
  );
};
