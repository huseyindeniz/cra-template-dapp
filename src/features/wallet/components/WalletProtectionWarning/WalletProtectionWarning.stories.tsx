// ConnectButton.stories.ts|tsx
import type { Meta, StoryObj } from '@storybook/react';

import { WalletProtectionWarning } from './WalletProtectionWarning';

const meta: Meta<typeof WalletProtectionWarning> = {
  component: WalletProtectionWarning,
};
export default meta;

type Story = StoryObj<typeof WalletProtectionWarning>;

export const Default: Story = { args: {} };
