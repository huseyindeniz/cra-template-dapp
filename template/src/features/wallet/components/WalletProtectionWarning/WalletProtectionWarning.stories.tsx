// ConnectButton.stories.ts|tsx
import { ComponentStory, ComponentMeta } from '@storybook/react';

import { WalletProtectionWarning } from './WalletProtectionWarning';

export default {
  title: 'wallet/WalletProtectionWarning',
  component: WalletProtectionWarning,
} as ComponentMeta<typeof WalletProtectionWarning>;

export const Default: ComponentStory<typeof WalletProtectionWarning> = args => (
  <WalletProtectionWarning />
);
