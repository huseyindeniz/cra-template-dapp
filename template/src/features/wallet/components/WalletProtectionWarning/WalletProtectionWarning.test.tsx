import { composeStories } from '@storybook/testing-react';
import { render } from '@testing-library/react';

import * as stories from './WalletProtectionWarning.stories';

describe('Feature: Wallet', () => {
  describe('Component: ConnectButton/WalletProtectionWarning', () => {
    const { Default } = composeStories(stories);
    describe('Scenario: Default', () => {
      it('should be visible and clickable', () => {
        // Arrange
        const expectedTitle = 'Access Denied';
        // Act
        const { asFragment, getByText } = render(<Default />);
        // Assert
        const actualTitle = getByText(expectedTitle);
        expect(actualTitle).toBeVisible();
        expect(asFragment).toMatchSnapshot();
      });
    });
  });
});
