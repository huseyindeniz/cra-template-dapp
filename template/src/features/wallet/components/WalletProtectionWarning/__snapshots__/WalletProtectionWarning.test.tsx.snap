// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Feature: Wallet Component: ConnectButton/WalletProtectionWarning Scenario: Default should be visible and clickable 1`] = `[Function]`;
